File: google-sheets-accessor-depr/app/enums/application_status.py
from enum import Enum


class ApplicationStatus(str, Enum):
    APPLIED = "applied"
    MEETING = "meeting"
    OFFER = "offer"
    DENIED = "denied"
    PENDING = "pending"

    @classmethod
    def is_valid(cls, value: str) -> bool:
        return value in cls._value2member_map_

    @classmethod
    def get_all(cls) -> list[str]:
        return [member.value for member in cls]

File: telegram-bot/internal/pkg/enums/applicationstatus.go
package enums

type ApplicationStatus string

const (
	ApplicationStatusApplied ApplicationStatus = "applied"
	ApplicationStatusMeeting ApplicationStatus = "meeting"
	ApplicationStatusOffer   ApplicationStatus = "offer"
	ApplicationStatusDenied  ApplicationStatus = "denied"
	ApplicationStatusPending ApplicationStatus = "pending"
)

var applicationStatusValues = []ApplicationStatus{
	ApplicationStatusApplied,
	ApplicationStatusMeeting,
	ApplicationStatusOffer,
	ApplicationStatusDenied,
	ApplicationStatusPending,
}

func (s ApplicationStatus) IsValid() bool {
	for _, status := range applicationStatusValues {
		if s == status {
			return true
		}
	}

	return false
}

func ApplicationStatusAll() []ApplicationStatus {
	result := make([]ApplicationStatus, len(applicationStatusValues))
	copy(result, applicationStatusValues)

	return result
}
